controllers/policies/clusteradmissionpolicystatus_controller.go

```diff
@@ -0,0 +1,185 @@
+		return ctrl.Result{}, fmt.Errorf("cannot retrieve policy server ConfigMap: %w", err)
+	}
+	policyMap, err := getPolicyMapFromConfigMap(&policyServerConfigMap)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot retrieve policy map from ConfigMap: %w", err)
+	}
+	if policy, ok := policyMap[req.Name]; ok {
+		clusterAdmissionPolicy.Status.PolicyMode = policiesv1alpha2.ClusterAdmissionPolicyModeStatusEnum(policy.PolicyMode)
+	} else {
+		clusterAdmissionPolicy.Status.PolicyMode = policiesv1alpha2.ClusterAdmissionPolicyModeStatusUnknown
+	}
+	policyServerName := fmt.Sprintf("policy-server-%s", clusterAdmissionPolicy.Spec.PolicyServer)
+	policyServerDeployment := appsv1.Deployment{}
+	if err := r.APIReader.Get(ctx, types.NamespacedName{Namespace: r.Reconciler.DeploymentsNamespace, Name: policyServerName}, &policyServerDeployment); err != nil {
+		return ctrl.Result{}, fmt.Errorf("cannot retrieve policy server ConfigMap: %w", err)
+	}
+	if configAnnotation, ok := policyServerDeployment.Annotations[constants.PolicyServerDeploymentConfigVersionAnnotation]; ok {
+		if configAnnotation == policyServerConfigMap.ResourceVersion {
+			apimeta.SetStatusCondition(
+				&clusterAdmissionPolicy.Status.Conditions,
+				metav1.Condition{
+					Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+					Status:  metav1.ConditionTrue,
+					Reason:  "ConfigurationVersionMatch",
+					Message: "Configuration for this policy is up to date",
+				},
+			)
+		} else {
+			apimeta.SetStatusCondition(
+				&clusterAdmissionPolicy.Status.Conditions,
+				metav1.Condition{
+					Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+					Status:  metav1.ConditionFalse,
+					Reason:  "ConfigurationVersionMismatch",
+					Message: "Configuration for this policy is not up to date",
+				},
+			)
+		}
+	} else {
+		apimeta.SetStatusCondition(
+			&clusterAdmissionPolicy.Status.Conditions,
+			metav1.Condition{
+				Type:    string(policiesv1alpha2.PolicyServerConfigurationUpToDate),
+				Status:  metav1.ConditionFalse,
+				Reason:  "UnknownConfigurationVersion",
+				Message: fmt.Sprintf("Configuration version annotation (%s) in deployment %s is missing", constants.PolicyServerDeploymentConfigVersionAnnotation, policyServerName),
+			},
+		)
+	}
+
+	// Update status
+	err = r.Status().Update(ctx, &clusterAdmissionPolicy, &client.UpdateOptions{})
+	return ctrl.Result{Requeue: true}, err
+}
+
+// SetupWithManager sets up the controller with the Manager.
+// nolint:wrapcheck
+func (r *ClusterAdmissionPolicyStatusReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&policiesv1alpha2.ClusterAdmissionPolicy{}).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(r.findClusterAdmissionPoliciesForConfigMap),
```

Comment:
have you considered using `CreateFunc`, `UpdateFunc` and `DeleteFunc` as we are doing here  https://github.com/kubewarden/kubewarden-controller/pull/173 instead of `EnqueueRequestsFromMapFunc`?
`EnqueueRequestsFromMapFunc` is called twice in updates and in my opinion is a bit less clear than the other funcs

Reply:
The goal behind `EnqueueRequestsFromMapFunc` here is that when a change happens in a ConfigMap that contains several policies in its policy map, we need to trigger N reconciliations (1 to N) because, given a configmap containing:

```
{"policy1":{"mode": "monitor"}, "policy2":{"mode": "monitor"}}
```

What should happen if it is changed to:

```
{"policy1":{"mode": "protect"}, "policy2":{"mode": "protect"}}
```

Is that reconciliations are requested for both `policy1` and `policy2`. Given this is a 1:N relationship, `EnqueueRequestsFromMapFunc` is the best fit to update policy statuses based on what is observed IMO.